core/trino-main/src/main/java/io/trino/execution/ColumnInfo.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import java.util.Optional;
+
+// XXX: ColumnInfo allows the grouping of all the naming elements necessary to deliver complete ResultSetMetaData (see issue#22306).
+public record ColumnInfo(Optional<String> catalog, Optional<String> schema, Optional<String> table, String name, Optional<String> label)
```

Comment:
Java record doesn't validate non-nullness of its components automatically. This needs to be done explicitly in each class.
see here for example
 https://github.com/trinodb/trino/blob/53cee48ae8b30368e8d8669786f000f44ed1f47c/core/trino-main/src/main/java/io/trino/json/ir/IrAbsMethod.java#L27